src/control/cmd/agent/mgmt_rpc.go

```diff
@@ -103,12 +149,25 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte) ([]byte, error) {
 		return nil, errors.Wrapf(err, "GetAttachInfo %s %v", mod.ap, *req)
 	}
 
-	mod.resmgmtpb, err = proto.Marshal(resp)
+	if resp.Provider == "" {
+		return nil, errors.Errorf("GetAttachInfo %s %v contained no provider.", mod.ap, *req)
+	}
+
+	// Scan the local fabric to determine what devices are available that match our provider
+	scanResults, err := netdetect.ScanFabric(resp.Provider)
 	if err != nil {
-		return nil, drpc.MarshalingFailure()
+		return nil, err
+	}
+
+	err = mod.aiCache.initResponseCache(resp, scanResults)
+	if err != nil {
+		return nil, err
 	}
 
-	mod.cachedAttachInfo = true
+	resmgmtpb, err := mod.aiCache.getResponse(numaNode)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Likewise here. You'd only need to check for non-nil err and explicitly return `nil, err` if your `getResponse()` method was pathological and returned something other than nil with an error.

Reply:
Good catch.